inf-ruby.el

```diff
@@ -452,11 +452,18 @@ Must not contain ruby meta characters.")
            ;; Evaluation seems to have failed.
            ;; Try to extract the error string.
            (let* ((inhibit-field-text-motion t)
-                  (s (buffer-substring-no-properties (point) (line-end-position))))
+                  (s (buffer-substring-no-properties (point)
+                                                     (progn
+                                                       (re-search-forward inf-ruby-prompt-pattern)
+                                                       (re-search-backward inf-ruby-prompt-pattern)
```

Comment:
Can you do `(goto-char (1- (match-beginning 0)))` here instead of the last two calls?